delayed_spans.go

```diff
@@ -0,0 +1,77 @@
+// (c) Copyright IBM Corp. 2022
+
+package instana
+
+import (
+	"net/url"
+)
+
+const maxDelayedSpans = 500
+
+var delayed = &delayedSpans{
+	spans: make(chan *spanS, maxDelayedSpans),
+}
+
+type delayedSpans struct {
+	spans chan *spanS
+}
+
+func (ds *delayedSpans) append(span *spanS) bool {
+	select {
+	case ds.spans <- span:
+		return true
+	default:
+		return false
+	}
+}
+
+func (ds *delayedSpans) flush() {
+	for {
+		select {
+		case s := <-ds.spans:
+			t, ok := s.Tracer().(Tracer)
+			if !ok {
+				continue
+			}
+
+			if err := ds.processSpan(s, t.Options()); err != nil {
+				continue
+			}
+
+			if sensor.Agent().Ready() {
+				s.tracer.recorder.RecordSpan(s)
+			} else {
+				ds.append(s)
+				return
+			}
+		default:
+			return
+		}
+	}
+}
+
+func (ds *delayedSpans) processSpan(s *spanS, opts TracerOptions) error {
```

Comment:
Suggestion: Addition of short function description as comment would have been nice

Reply:
Done